Sources/Cryptor/Digest.swift

```diff
@@ -87,59 +101,79 @@ public class Digest: Updatable {
         switch algorithm {
 			
         case .md2:
-			#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
-	            engine = DigestEngineCC<CC_MD2_CTX>(initializer:CC_MD2_Init, updater:CC_MD2_Update, finalizer:CC_MD2_Final, length:CC_MD2_DIGEST_LENGTH)
-			#elseif os(Linux)
+			#if os(Linux)
 				fatalError("MD2 digest not supported by OpenSSL")
+            #elseif os(iOS)
+                if #available(iOS 13, *) {
+                    fatalError("MD2 digest is cryptographically broken")
+                } else {
+                    self.engine = DigestEngineCC<CC_MD2_CTX>(initializer:CC_MD2_Init, updater:CC_MD2_Update, finalizer:CC_MD2_Final, length:CC_MD2_DIGEST_LENGTH)
+                }
+            #else
+                self.engine = DigestEngineCC<CC_MD2_CTX>(initializer:CC_MD2_Init, updater:CC_MD2_Update, finalizer:CC_MD2_Final, length:CC_MD2_DIGEST_LENGTH)
 			#endif
 			
         case .md4:
-			#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
-            	engine = DigestEngineCC<CC_MD4_CTX>(initializer:CC_MD4_Init, updater:CC_MD4_Update, finalizer:CC_MD4_Final, length:CC_MD4_DIGEST_LENGTH)
-			#elseif os(Linux)
-				engine = DigestEngineCC<MD4_CTX>(initializer:MD4_Init, updater:MD4_Update, finalizer:MD4_Final, length:MD4_DIGEST_LENGTH)
+            #if os(Linux)
+                self.engine = DigestEngineCC<MD4_CTX>(initializer:MD4_Init, updater:MD4_Update, finalizer:MD4_Final, length:MD4_DIGEST_LENGTH)
+            #elseif os(iOS)
+                if #available(iOS 13, *) {
+                    fatalError("MD4 digest is cryptographically broken")
```

Comment:
Seems like the failure here is more that MD4 is not supported on iOS 13 and above, not that it is insecure?

Reply:
It is still supported (as far as I can tell), it just generates a build warning (guessing that may mean that soon it will stop being supported).